src/test/operators/table_scan_test.cpp

```diff
@@ -546,10 +552,27 @@ TEST_F(OperatorsTableScanTest, ScanForNullValuesWithNullRowIDOnReferencedDictCol
   auto table_wrapper = std::make_shared<TableWrapper>(table);
   table_wrapper->execute();
 
-  const auto tests = std::map<ScanType, std::vector<AllTypeVariant>>{{ScanType::OpEquals, {123, 1234}},
-                                                                     {ScanType::OpNotEquals, {12345, NULL_VALUE}}};
+  const auto tests = std::map<ScanType, std::vector<AllTypeVariant>>{{ScanType::OpIsNull, {123, 1234}},
+                                                                     {ScanType::OpIsNotNull, {12345, NULL_VALUE}}};
 
   scan_for_null_values(table_wrapper, tests);
 }
 
+TEST_F(OperatorsTableScanTest, NullSemantics) {
+  const auto scan_types =
+      std::vector<ScanType>({ScanType::OpEquals, ScanType::OpNotEquals, ScanType::OpLessThan,
+                             ScanType::OpLessThanEquals, ScanType::OpGreaterThan, ScanType::OpGreaterThanEquals});
+
+  for (auto scan_type : scan_types) {
+    auto scan = std::make_shared<TableScan>(_table_wrapper, ColumnID{0}, scan_type, NULL_VALUE);
```

Comment:
int_float.tbl does not have any NULL values. This does not test that `= NULL` is false even for a NULL value, does it?

Reply:
You're right. fixed